### DISCUSSION TITLE
PDF file doesn't load in Vercel-Flask Webpage

### DISCUSSION THREAD

**Discussion by Bibhav48** [vercel/community#4010] [Bug Report]
### Page to Investigate

https://notesninja.vercel.app/notes

### Steps to Reproduce

I am using Flask where `vercel.json` is:
```json
{
    "version": 2,
    "builds": [
        {
            "src": "app.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "/app.py"
        }
    ]
}
```
The PDF's location is accurate it works when I run it locally in every different browser and operating system. The problem is when I load in Vercel it has a 404 error. The `css` file in the same `\static` folder loads but `pdf` file doesn't.

Main code:
```html
<a href ="{{ link }}" target="_blank">
```
where link is `static/notes/physics/2 Motion.pdf` but I also have tried `<a href ="{{ url_for('static',filename='2 Motion.pdf') }}">` to check if nested folder was causing problem.

Also to confirm I checked it with other hosts. It worked there but not in Vercel. Although I used `gunicorn` in the other host, I assume it must not be a problem just to load a pdf

**Comment by amyegan** [comment]
Hey @Bibhav48. I wasn't able to replicate the problem in my test app. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share to give us a better look at what's going wrong?

**Reply by Bibhav48** [reply]
@amyegan the custom `404.html` is what I see on clicking the link.
**And here is MRE:**
sample dir:
```
templates/
├─ notes.html
├─ 404.html
static/
├─ notes/
│  ├─ physics/
│  │  ├─ 1 Introduction.pdf
│  │  ├─ 2 Other.pdf
app.py
requirements.txt
vercel.json
```

requirements.txt
```
Flask
gunicorn
```
app.py
```python
#all imports rest of routes
#...
@app.route('/notes')
def notes():
    files =[]
    folder_path = "static/notes/physics"
    if os.path.exists(folder_path):
        file_names = sorted(os.listdir(folder_path))
        files = [(i," ".join(f.split(" ")[1::]).split(".")[0]) for i,f in enumerate(file_names,start=1)]
        links = [f"{folder_path}/{file}" for file in file_names]
        files_phy=zip(files,links)
    return render_template("notes.html", active="notes",files_phy=files_phy)

@app.errorhandler(404)
def not_found(e):
    return render_template("404.html")
```
index.html
```html
<!-- bootstrap and CSS links and rest code -->

          <div class="accordion-body">
            <div class="list-unstyled">
              {% for (index,file),link in  files_phy%}
              <div class="custom-pdf-icon">
                <a href ="{{ link }}" target="_blank">
                  <i class="fa fa-file-pdf-o"></i>
                  <span class="pdf-text">{{file}}</span>
                </a>
              </div>
              {% endfor%}
```
What I thought was: If the link is correct `dir` is proper and the link to dir is valid, It runs in my local machine and another test host site, so why not here? So I thought it must be my configuration in`vercel.json` or with `vercel` itself. Now I am even more confused. I hope this will be solved.

**Reply by amyegan** [reply]
Can share this code as a runnable, public, [MRE](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) repo? That would be very helpful for collaborative debugging!

**Reply by Bibhav48** [reply]
@amyegan this is my MRE [repo](https://github.com/Bibhav48/test-repo) and link to its vercel [web-page](https://test-repo-zeta-puce.vercel.app/notes) and the error [link](https://test-repo-zeta-puce.vercel.app/static/notes/physics/1%20Introduction.pdf).

**Reply by Bibhav48** [reply]
@amyegan I  think I have found the cause. The file named with spaces doesn't load. For eg: `"1 Indtroduction.pdf"` is not loaded but,`"1Introduction.pdf"` and `"Introduction.pdf"` loads. But why so? It seems to be a problem in Vercel or Vercel with Flask.
Also now I have updated the [MRE](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) repo with different naming styles: one with space doesn't work but others do.

**Reply by Bibhav48** [reply]
After a bit of digging, I also found an [issue](https://github.com/vercel/next.js/issues/37273) from 2022 that is similar to mine. The issue was fixed for `next.js`. The issue seems to still exist for `Flask`.
